extension/encoding/jsonlogencodingextension/extension.go

```diff
@@ -68,3 +71,48 @@ func (e *jsonLogExtension) Start(_ context.Context, _ component.Host) error {
 func (e *jsonLogExtension) Shutdown(_ context.Context) error {
 	return nil
 }
+
+func (e *jsonLogExtension) LogProcessor(ld plog.Logs) ([]byte, error) {
+	prettyLogs := []prettyLogBody{}
+
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		resourceAttrs := attrsValue(rl.Resource().Attributes())
+
+		sls := rl.ScopeLogs()
+		for j := 0; j < sls.Len(); j++ {
+			sl := sls.At(j)
+			logs := sl.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				log := logs.At(k)
+				logEvent := prettyLogBody{
+					Body:       log.Body().AsRaw(),
+					Resource:   resourceAttrs,
+					Attributes: attrsValue(log.Attributes()),
+				}
+				prettyLogs = append(prettyLogs, logEvent)
+			}
+		}
+	}
+
+	return jsoniter.Marshal(prettyLogs)
+}
+
+type prettyLogBody struct {
+	Body       any            `json:"body,omitempty"`
+	Attributes map[string]any `json:"attributes,omitempty"`
+	Resource   map[string]any `json:"resource,omitempty"`
```

Comment:
Maybe `log_attributes` and `resource_attributes`?

Reply:
I was thinking of trying to stay with camelCase as most json is but I wouldn't be opposed to `logAttributes` and `resourceAttributes`